SU2_CFD/include/numerics_structure.hpp

```diff
@@ -2071,34 +2071,64 @@ class CUpwLin_AdjTurb : public CNumerics {
 };
 
 /*!
- * \class CUpwSca_TurbSA
- * \brief Class for doing a scalar upwind solver for the Spalar-Allmaral turbulence model equations.
+ * \class CUpwSca_Scalar
+ * \brief Template class for scalar upwind fluxes between nodes i and j.
+ * \details This class serves as a template for the scalar upwinding residual
+ *   classes.  The general structure of a scalar upwinding calculation is the
+ *   same for many different  models, which leads to a lot of repeated code.
+ *   By using the template design pattern, these sections of repeated code are
+ *   moved to this shared base class, and the specifics of each model
+ *   are implemented by derived classes.  In order to add a new residual
+ *   calculation for a convection residual, extend this class and implement
+ *   the pure virtual functions with model-specific behavior.
  * \ingroup ConvDiscr
- * \author A. Bueno.
+ * \author C. Pederson, A. Bueno., and A. Campos.
  * \version 5.0.0 "Raven"
  */
-class CUpwSca_TurbSA : public CNumerics {
+class CUpwSca_Scalar : public CNumerics {
```

Comment:
For me, CUpwSca_Scalar is a bit redundant... CUpw_Scalar or CUpwScalar might be better. Perhaps the latter is best, as later child classes can be more specific to the model, e.g., CUpwScalar_TurbSA, etc.